pkg/identity/identity.go

```diff
@@ -149,15 +150,24 @@ func PeerIdentityFromContext(ctx context.Context) (*PeerIdentity, error) {
 	return PeerIdentityFromPeer(p)
 }
 
-// NodeIDFromKey hashes a publc key and creates a node ID from it
+// NodeIDFromKey hashes a public key and creates a node ID from it
 func NodeIDFromKey(k crypto.PublicKey) (storj.NodeID, error) {
+	if ek, ok := k.(*ecdsa.PublicKey); ok {
+		return NodeIDFromECDSAKey(ek)
+	}
+	return storj.NodeID{}, storj.ErrNodeID.New("invalid key type: %T", k)
+}
+
+// NodeIDFromECDSAKey hashes a public key and creates a node ID from it
+func NodeIDFromECDSAKey(k *ecdsa.PublicKey) (storj.NodeID, error) {
+	// id = sha256(sha256(pkix(k)))
 	kb, err := x509.MarshalPKIXPublicKey(k)
 	if err != nil {
 		return storj.NodeID{}, storj.ErrNodeID.Wrap(err)
 	}
-	hash := make([]byte, len(storj.NodeID{}))
-	sha3.ShakeSum256(hash, kb)
-	return storj.NodeIDFromBytes(hash)
+	mid := sha256.Sum256(kb)
+	end := sha256.Sum256(mid[:])
```

Comment:
What's going on here?

Reply:
just doin what satoshi did: https://bitcoin.stackexchange.com/questions/8443/where-is-double-hashing-performed-in-bitcoin. seems like it worked out okay. i can't talk myself out of being worried about hash length extension attacks myself